streamalert/classifier/clients/firehose.py

```diff
@@ -52,19 +53,32 @@ class FirehoseClient:
     MAX_RECORD_SIZE = 1000 * 1000 - 2
 
     # Default firehose name format, should be formatted with deployment prefix
-    DEFAULT_FIREHOSE_FMT = '{}streamalert_data_{}'
+    DEFAULT_FIREHOSE_FMT = '{}data_{}'
```

Comment:
IMO I'd rather keep the `streamalert` portion of this and drop the `data` bit.. these are `Firehose Data Delivery Streams` and including `data` is pretty redundant. at least having `streamalert` in the name will help with things like filtering streams in console, etc

Reply:
I mentioned this before, but I recommend explicitly stating the firehose names in the `conf/` directory, and explicitly populating them during the `generate` step. I imagine the workflow is like:

* Look in conf directory for explicit firehose name
* If no explicit name, use `use_prefix` and the log type
* During `manage.py generate`, explicitly populate into Terraform json files
* Instead of generating them using `locals {$prefix}_data_{$hash}`, just use the explicit name

Reply:
@Ryxias quick question.
> Look in conf directory for explicit firehose name

Potentially, we may have hundreds of firehose up running due to one log schema using one firehose. The firehose conf will be quite long and add complexity to read all firehose names from the `conf/global.json`.  IMO, I would take @ryandeivert 's suggestion. Thoughts?

Reply:
I do like Derek's idea, I'm just unsure where this would live.. in `global.json`? how do we map from `logs.json` to this reliably... or does it live in `logs.json`?? any suggestions for _how_ this would actually look?

Reply:
I also **still** insist we drop `_data_`, even with the hashing route. really not sure why we're hung up on keeping it...

Reply:
Doing it my way will not require a re-deploy of Firehoses, as you can simply specify the original Firehose name.

But if you **need** to redeploy the Firehoses anyway because of the Terraform upgrade from `s3` to `extended_s3`, then I suppose it's fine to do it your way.

Reply:
I think having to manually specify all of the firehose names for each schema is going to take as much time if not more, than to just recreate the firehoses. Also v3 technically isn't "production" yet so I think a breaking change is ok.

Reply:
@Ryxias you are right, the firehoses will be recreated when they are upgraded from `s3` to `extended_s3`.
Since @Ryxias 's idea is getting popular, we probably can have firehose custom name in the `conf/schems/*.json` because firehose name is tightly linked to schema name.

Reply:
@Ryxias again I'm not opposed to your way, I just want to hear more about "how" this will live in the config. does it go in logs.json or the firehose config of global.json?

Reply:
We can have it in `conf/logs.json`, or `conf/schemas/*.json or `conf/firehose.json`. I don't really care either way. 

I think the important one is that it's OPTIONAL. In most cases, you **_should not explicitly specify_** your firehoses. But they should be explicitly stated somewhere. I don't care much about the `conf/****.json` one, but I am very strongly in favor of explicitly generating them in the `terraform/***.tf.json` files, instead of dynamically creating them inside of the the Terraform modules.

Reply:
It would be mean your `manage.py generate` step would just have a new function that's like:


```
def get_firehose_name(log_type):
  # Explicitly stated
  if config['whatever'].get(log_type, {}).get('firehose_name', None):
    return config['whatever'].get(log_type, {}).get('firehose_name', None)

  # Generated
  use_prefix = config['global']['whatever']['use_prefix']

  firehose_name = '{}{}_{}'.format(
    use_prefix ? '{}_'.format(prefix),
    'streamalert_data' # Or whatever
    log_schema
  )

  if len(firehose_name) > 64:
    # then do some hashing stuff

  return firehose_name
```

Reply:
I'm drawing attention to the _where_ on this because I think we're forgetting how this is currently performed. data retention is _optional_ and does not need to be enabled.. but enabling means toggling it in `global.json`. having _some_ settings in global.json and others in logs.json is confusing.. but on the other hand, having the name for an optional feature embedded in logs.json and other settings in global.json is also confusing. we're sorta asking a lot of users here to mind-map all of this together

Reply:
Wrap up this thread. I took Ryan's suggestion and addressed it in [this commit](https://github.com/airbnb/streamalert/pull/1191/commits/d4cd2d3493a4f0d883d7702f33de6c766c28fe71). We will support the custom firehose stream name later and I have open an issue #1193 to track the future work.